script/split_tests.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+"""Helper script to split test into n buckets."""
+
+from __future__ import annotations
+
+import argparse
+from dataclasses import dataclass, field
+from math import ceil
+import os
+import subprocess
+import sys
+
+
+class Bucket:
+    """Class to hold bucket."""
+
+    def __init__(
+        self,
+    ):
+        """Initialize bucket."""
+        self.tests = 0
+        self._paths = []
+
+    def add(self, part: TestFolder | TestFile):
+        """Add tests to bucket."""
+        self.tests += part.total_tests
```

Comment:
This makes a copy of `self.tests`. If you want to reuse the original list, you should use `self.tests.extend(part.total_tests)`